cg/meta/demultiplex/status_db_storage_functions.py

```diff
@@ -75,10 +70,96 @@ def store_sequencing_metrics_in_status_db(flow_cell: FlowCellDirectoryData, stor
     add_sequencing_metrics_to_statusdb(
         sample_lane_sequencing_metrics=sample_lane_sequencing_metrics, store=store
     )
-
+    store_metrics_for_non_pooled_undetermined_reads(flow_cell=flow_cell, store=store)
     LOG.info(f"Added sequencing metrics to status db for: {flow_cell.id}")
 
 
+def store_metrics_for_non_pooled_undetermined_reads(
+    flow_cell: FlowCellDirectoryData, store: Store
+) -> None:
+    """Store sequencing metrics for non pooled undetermined reads in status db."""
+    non_pooled_lanes_and_samples: List[
+        Tuple[int, str]
+    ] = flow_cell.sample_sheet.get_non_pooled_lane_sample_id_pairs()
+
+    metrics: List[
+        SampleLaneSequencingMetrics
+    ] = create_undetermined_sequencing_metrics_for_flow_cell(
+        flow_cell_directory=flow_cell.path,
+        bcl_converter=flow_cell.bcl_converter,
+        non_pooled_lanes_and_samples=non_pooled_lanes_and_samples,
+    )
+
+    for undetermined_metric in metrics:
+        sample_id: str = undetermined_metric.sample_internal_id
+        lane: int = undetermined_metric.flow_cell_lane_number
+
+        existing_metric: SampleLaneSequencingMetrics = (
+            store.get_metrics_entry_by_flow_cell_name_sample_internal_id_and_lane(
+                flow_cell_name=flow_cell.id, sample_internal_id=sample_id, lane=lane
+            )
+        )
+        if existing_metric:
+            update_existing_metric(existing_metric=existing_metric, new_metric=undetermined_metric)
+        else:
+            store.session.add(undetermined_metric)
```

Comment:
Might be necessary to add check whether the associated sample exists in status db before tracking the new metric